### DISCUSSION TITLE
Is it possible to make the editor not re-render when the preview is shown

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#7745] [General]
In my codemirror-6 plugin I'm looking for a way to set back focus to the editor when the preview is shown or hidden
To do so, it's necessary that the editor doesn't get redrawn when the preview is shown
I know, there would be some changes in the core to be made for this to be possilbe
But is it even possible?

I'm thinking of ways and trying but couldn't find a way yet

**Comment by BurningTreeC** [comment]
Ah, I found a way 😄 

Edit: no, I didn't

**Comment by BurningTreeC** [comment]
I would like to wrap the editor and the preview together in a div ... 🤔

**Reply by pmario** [reply]
And that's probably the problem. You need to split them into different widget trees. So they can be redrawn independently. Otherwise the core redraw mechanism will "bite" you.

**Reply by BurningTreeC** [reply]
Thanks @pmario, I tried but I don't see a solution :(

**Comment by Jermolene** [comment]
Hi @BurningTreeC could you test the quick fix at #7747?

**Reply by BurningTreeC** [reply]
Hi @Jermolene , yes I've tested this and it works, thank you very much!

That leaves one problem open:

I'd like to wrap the editor and the preview in one div that will have `display: flex`
That means, the editor toolbar must be above that div
Is that possible?

**Reply by BurningTreeC** [reply]
Ah, and another thing @Jermolene 

The preview button needs a tm-edit-text-operation message "focus-editor"

**Reply by Jermolene** [reply]
> I'd like to wrap the editor and the preview in one div that will have `display: flex` That means, the editor toolbar must be above that div Is that possible?

That would require a deep change to the edit widget. At the moment, the toolbar is a sibling of the editor in the DOM, but a child of the edit widget in the widget tree. The way to pursue it would perhaps be to first develop a way for the editor toolbar to be independent of the editor itself, then we'll have a lot more freedom in how things are laid out.

> The preview button needs a tm-edit-text-operation message "focus-editor"

That's a good idea thank you, added in bb3253fb2118a802d7c2b42725de524a3fa0b90b

**Reply by BurningTreeC** [reply]
> That would require a deep change to the edit widget. At the moment, the toolbar is a sibling of the editor in the DOM, but a child of the edit widget in the widget tree. The way to pursue it would perhaps be to first develop a way for the editor toolbar to be independent of the editor itself, then we'll have a lot more freedom in how things are laid out.

Yes I know, it would need a deep change.

Thank you for creating that branch! 👍 
I'll test it!

**Reply by BurningTreeC** [reply]
The fix-body-editor-refresh branch works beautifully. Thank you @Jermolene
